File: UnityProject/BrainAcademy/Assets/Scripts/Models/GameModels.cs
using System.Collections.Generic;

[System.Serializable]
public class Question
{
    public string label;
    public string questionText;
    public List<string> sequence;
    public List<string> answers;
    public string correctAnswer;
    public string explanation;

    public Question(string label, List<string> answers, string correctAnswer,
        string questionText = null, List<string> sequence = null,
        string explanation = null)
    {
        this.label = label;
        this.questionText = questionText;
        this.sequence = sequence;
        this.answers = answers;
        this.correctAnswer = correctAnswer;
        this.explanation = explanation;
    }
}

public abstract class GameQuestion
{
    public class Standard : GameQuestion
    {
        public Question question;
        public Standard(Question question) { this.question = question; }
    }

    public class ReadingComprehension : GameQuestion
    {
        public string passageId;
        public string passageTitle;
        public string passageText;
        public string passageType;
        public Question question;

        public ReadingComprehension(string passageId, string passageTitle,
            string passageText, string passageType, Question question)
        {
            this.passageId = passageId;
            this.passageTitle = passageTitle;
            this.passageText = passageText;
            this.passageType = passageType;
            this.question = question;
        }
    }
}

[System.Serializable]
public class GameStats
{
    public int totalScore;
    public int gamesPlayed;
    public int bestStreak;
}

File: app/src/main/java/com/braincademy/junior/model/GameModels.kt
package com.braincademy.junior.model

enum class Category(val displayName: String) {
    MATH("Math"),
    LOGIC("Logic & Puzzles"),
}

enum class Difficulty(val displayName: String, val points: Int, val timerSeconds: Int) {
    EASY("Easy", 10, 20),
    MEDIUM("Medium", 25, 15),
    HARD("Hard", 50, 10),
    SUPER_HARD("Super Hard", 100, 7),
}

data class Question(
    val label: String,
    val questionText: String? = null,
    val sequence: List<String>? = null,
    val answers: List<String>,
    val correctAnswer: String,
)

data class MemoryQuestion(
    val gridSize: Int,
    val highlighted: List<Int>,
    val showTimeMs: Long,
)

sealed class GameQuestion {
    data class Standard(val question: Question) : GameQuestion()

    data class Memory(val memoryQuestion: MemoryQuestion) : GameQuestion()
}

data class GameStats(
    val totalScore: Int = 0,
    val gamesPlayed: Int = 0,
    val bestStreak: Int = 0,
)
